addons/account_edi_peppol/models/account_edi_format.py

```diff
@@ -0,0 +1,266 @@
+from odoo.osv import expression
+
+
+class AccountEdiFormat(models.Model):
+    _inherit = 'account.edi.format'
+
+    # -------------------------------------------------------------------------
+    # Export
+    # -------------------------------------------------------------------------
+
+    def _check_for_incompatibilities(self, invoice):
+        '''Throws an error if some information is missing to generate valid UBL'''
+
+        if not invoice.company_id.partner_id.commercial_partner_id.peppol_endpoint_scheme:
+            return _('Only companies with a European VAT are supported')
+            # See PEPPOL_COUNTRY_EAS : only those VAT number are supported.
+            # Other identification are possible but not yet implemented in Odoo, see : https://docs.peppol.eu/poacc/billing/3.0/codelist/eas/
+        if not invoice.company_id.partner_id.commercial_partner_id.peppol_endpoint:
+            return _('Please set a vat number on the seller\'s company')
+
+        if not invoice.commercial_partner_id.peppol_endpoint_scheme:
+            return _('Only companies with a European VAT are supported (customer)')
+        if not invoice.commercial_partner_id.peppol_endpoint:
+            return _('Please set a vat number on the customer\'s company')
+        return False
+
+    def _export_peppol(self, invoice):
+        self.ensure_one()
+
+        def format_monetary(amount, currency=invoice.currency_id):
+            # Format the monetary values to avoid trailing decimals (e.g. 90.85000000000001).
+            return float_repr(amount, currency.decimal_places)
+
+        def convert_monetary(amount, from_currency):
+            # All monetary should be in the invoice currency, except for vat total
+            return from_currency._convert(amount, invoice.currency_id, invoice.company_id, invoice.invoice_date)
+
+        def get_tax_total():
+            breakdown = {}
+            for line in invoice.invoice_line_ids:
+                line_taxes = line.tax_ids.compute_all(
+                    line.price_unit,
+                    quantity=line.quantity,
+                    product=line.product_id,
+                    partner=invoice.partner_id)['taxes']
+                currency = line.currency_id or invoice.currency_id
+                if line_taxes:
+                    for tax in line_taxes:
+                        tax_category = 'S' if tax['amount'] else 'Z'
+                        tax_percent = self.env['account.tax'].browse(tax['id']).amount
+                    if (tax_category, tax_percent) not in breakdown:
+                        breakdown[(tax_category, tax_percent)] = {'base': 0, 'amount': 0}
+                    breakdown[(tax_category, tax_percent)]['base'] = currency._convert(tax['base'], invoice.currency_id, invoice.company_id, invoice.invoice_date)
+                    breakdown[(tax_category, tax_percent)]['amount'] = currency._convert(tax['amount'], invoice.currency_id, invoice.company_id, invoice.invoice_date)
```

Comment:
conversion is not needed since the result of `compute_all` is already using the invoice's currency